### ISSUE TITLE
VIA not working for Nuphy air60 V2

### ISSUE THREAD

**Issue by sinail1** [the-via/releases#280]
So I purchased a 60% keyboard because it had arrow keys and the Shift_R key above the Left key and Delete key above the right key. I need to remap the Shift_R and Delete keys to be Prior (page up) and Next (page down) keys. 

I tried this with via-3.0.0-linux.deb and although the keys remap as I assign them to on the software, it doesn't seem to be actually remapping. Just to make sure, I switched the Q key to the A key and that doesn't seem to work either.

I should mention that I was having issues using VIA on the web: "Received invalid protocol version from device" and "Failed to open the device" were the error messages that I kept getting. 

VIA-3.0.0 is detecting my keyboard just fine (it says "NUPHY AIR60 V2" at the top right). Can I please get some help?

**Comment by kespii** [comment]
same for mine air60 v2

**Comment by dvladimirov** [comment]
same issue here am using the provided https://nuphy.com/pages/json-files-for-nuphy-keyboards starter config from their site and when i try to remap caps-lock to esc or any other key nothing happens. I tried installing manually qmk software to do a manual build(failed). Tried the local via app and the electron app in the browser all act the same way as OP describes.

**Comment by dvladimirov** [comment]
Tldr - contacted Nuphy directly and they let me know on a little secret.
If you are on Mac  - remap the keys in layer 0 and 1.
If you are on Win - remap the keys in layer 2 or 3.

And everything worked out great. Hope this helps someone.

**Comment by Azytzeen** [comment]
Thank you i will check it.
I have the same issue, waiting for a response from support atm, how about linux, did they tell on which layer remap a keys?

**Comment by Azytzeen** [comment]
> Thank you i will check it. I have the same issue, waiting for a response from support atm, how about linux, did they tell on which layer remap a keys?

I checked and on Win OS it worked from 3rd layer so 3rd layer is your first layer. This is the same for Linux, the same setup. 
Thanks

**Comment by cdzett** [comment]
> Tldr - contacted Nuphy directly and they let me know on a little secret. If you are on Mac - remap the keys in layer 0 and 1. If you are on Win - remap the keys in layer 2 or 3.
> 
> And everything worked out great. Hope this helps someone.

Thanks a lot. That was unexpected...

**Comment by prudnikov** [comment]
Does not work for me as well. Clicking "Authorize device" button, selecting keyboard (it is marked as paired), and nothing happens.

**Comment by louia** [comment]
Same here with air96 v2 with latest firmware 2.1.0 on via brave macos

**Comment by deusxanime** [comment]
> Tldr - contacted Nuphy directly and they let me know on a little secret. If you are on Mac - remap the keys in layer 0 and 1. If you are on Win - remap the keys in layer 2 or 3.
> 
> And everything worked out great. Hope this helps someone.

Just to add to this... I just got an Air60 v2 and it is slightly different. Not sure if you had a different model or version here that you were referencing or if they changed or updated something since, but layers are slightly different than you have here on my Air60 v2.

Switch to "Mac" = Layer 0, 1, 2 (default is layer 0, which then references layer 2/3 in the default mappings)
Switch to "Win" = Layer 3, 4, 5 (default is layer 3, which then references layer 4/5 in the default mappings)
Layer 6 = referenced in mappings on both layer 1 and 4, basically this layer just has a few keys that alter settings for the side LEDs so it is not OS specific 
Layer 7 = completely blank by default, so you can do whatever you want with it

So basically all the Mac/Win switch does is set your default layer to either 0 or 3, respectively, from what I see. After that it is all in how you map your keys and you can reference any layer you want from there (even crossover between the Mac/Win layers).

Thanks so much for this though, I was altering stuff in Layer 0 on my Windows system (with the switch set to "Win" of course) and nothing was changing on my kb and it confusing the heck out of me! After reading this and as soon as I started making my changes on Layer 3, it all made sense!
